components/windows_recall/windows_recall.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/windows_recall/windows_recall.h"
+
+#include <memory>
+
+#include "base/supports_user_data.h"
+#include "base/win/windows_version.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+
+namespace {
+
+struct WindowsRecallData : public base::SupportsUserData::Data {
```

Comment:
I don't understand why this is needed

Reply:
I have looked through the code and it appears that it should apply to each new render frame that is created. In any case this does not necessarily correlate to a restart if you have multiple profiles.